wemake_python_styleguide/visitors/tokenize/comments.py

```diff
@@ -98,6 +99,79 @@ def _post_visit(self) -> None:
             )
 
 
+@final
+class EmptyCommentVisitor(BaseTokenVisitor):
+    """Checks empty comment tokens."""
+
+    def __init__(self, *args, **kwargs) -> None:
+        """Initializes fields to track empty comments."""
+        super().__init__(*args, **kwargs)
+
+        self._prev_comment_line_num = -1
+        self._prev_non_empty = -1
+        self._in_same_block = True
+        self._block_alerted = False
+        self._has_reserved = False
```

Comment:
Where is `_reserved_token` defined/declared? It's good practice to define/declare all variables modified in methods etc in `__init__`

Reply:
I see. I just thought initializing it with a token object is strange and setting it to ``None`` is also prohibited. Anyway, I still added back the initialization.

Reply:
It's just a quick thought, but [`TokenInfo`](https://github.com/python/cpython/blob/cfb2f796342c224599eeab847e8593343a775408/Lib/tokenize.py#L45) is `namedtuple`, that means it's immutable, so we can define some constant like `EMPTY_TOKEN` or `SENTINEL_TOKEN` with some "empty"/"dummy" `TokenInfo` value, that shouldn't be created normally by `tokenize` and assign it to `_reserved_token` in `__init__`.
Maybe this will also make `_has_reserved` optional because you can always compare `_reserved_token` with `EMPTY_TOKEN`.

Reply:
I like that idea! It seems better for clarity and readability. I'll look into that in a bit.